src/main/java/io/jenkins/plugins/opentelemetry/backend/ObservabilityBackend.java

```diff
@@ -217,4 +217,9 @@ public int compareTo(ObservabilityBackendDescriptor o) {
             throw new IllegalStateException("failure to create error template");
         }
     }
+
+    // pulled up common method to super class.
+    public int generateHashCode(Object... objects) {
```

Comment:
this HashCode implementation is too general, you lose the control over the hascode generator and will ends in a error-prone hashcode generator, so a wrong match on equals comparisons.

Reply:
Same as @kuisathaverat , I don't perceive this change as a beautification of our code.

Reply:
Thanks for reviewing my changes. I see the reason why these refactorings do not help. I am closing the pull request.